src/main/java/com/meilisearch/sdk/http/ApacheHttpClient.java

```diff
@@ -49,29 +48,28 @@ private HttpResponse<?> execute(HttpRequest<?> request) throws Exception {
                 getCallback(response));
         try {
             return response.thenApply(this::mapResponse).get();
-        } catch (CancellationException | ExecutionException e) {
-            // todo: throw dedicated exception
-            throw new Exception(e);
+        } catch (Exception e) {
+            throw new MeilisearchException(e);
```

Comment:
Maybe use a `MeilisearchApiException` here?

Reply:
I'm not sure about this comment anymore to be honest

Reply:
The `ApacheHttpClient` will be removed so I admit that I didn't spend too much time on it